WordPress/src/main/java/org/wordpress/android/ui/reader/models/ReaderRelatedPost.java

```diff
@@ -3,84 +3,109 @@
 import android.support.annotation.NonNull;
 import android.text.TextUtils;
 
-import org.wordpress.android.models.ReaderPost;
-import org.wordpress.android.util.UrlUtils;
+import org.json.JSONObject;
+import org.wordpress.android.util.HtmlUtils;
+import org.wordpress.android.util.JSONUtils;
 
 /**
  * simplified version of a reader post that contains only the fields necessary for a related post
  */
 public class ReaderRelatedPost {
-    private final long mPostId;
-    private final long mBlogId;
-    private final String mTitle;
-    private final String mByline;
-    private final String mFeaturedImage;
-
-    public ReaderRelatedPost(@NonNull ReaderPost post) {
-        mPostId = post.postId;
-        mBlogId = post.blogId;
-
-        mTitle = post.getTitle();
-        mFeaturedImage = post.getFeaturedImage();
-
-        /*
-         * we want to include the blog name in the byline when it's available, and most sites
-         * will have a name, but in rare cases there isn't one so we show the domain instead
-         */
-        String blogNameOrDomain;
-        boolean hasBlogNameOrDomain;
-        if (post.hasBlogName()) {
-            blogNameOrDomain = post.getBlogName();
-            hasBlogNameOrDomain = true;
-        } else if (post.hasBlogUrl()) {
-            blogNameOrDomain = UrlUtils.getHost(post.getBlogUrl());
-            hasBlogNameOrDomain = true;
-        } else {
-            blogNameOrDomain = null;
-            hasBlogNameOrDomain = false;
+    private long mPostId;
+    private long mSiteId;
+    private boolean mIsFollowing;
+
+    private String mTitle;
+    private String mAuthorName;
+    private String mAuthorAvatarUrl;
+    private String mExcerpt;
+    private String mSiteName;
+    private String mFeaturedImageUrl;
+
+    // these are the specific fields we should ask for when requesting related posts from
+    // the endpoint - note that we want to avoid ever requesting the post content, since
+    // that makes the call much heavier
+    public static final String RELATED_POST_FIELDS = "ID,site_ID,title,excerpt,site_name,is_following,author,featured_image,featured_media";
+
+    public static ReaderRelatedPost fromJson(@NonNull JSONObject json) {
+        ReaderRelatedPost post = new ReaderRelatedPost();
+
+        post.mPostId = json.optLong("ID");
+        post.mSiteId = json.optLong("site_ID");
```

Comment:
I've favored using the various `optXxx` methods throughout the reader, but I'm certainly open to alternatives. How would you suggest I handle the errors?